src/HassClient/Client/HassClient.cs

```diff
@@ -225,6 +249,8 @@ public class HassClient : IHassClient
         /// </summary>
         private bool _isClosed;
 
+        private bool _subscriberToEventChannelExists;

```

Comment:
This bool is a bit strange, as I understand it calling ReadEventAsync once makes the observables start producing items.

Reply:
Maybe If you use the Subject class you can use its HasObservers property to see if events need to be passed (not sure if that is what you need here

Reply:
Ah so the problem is that you need a way to subscribe to events AND specify which events. 

Subscribing to events is actually what IObservable.Subscribe is for, but it does not allow you to specify which events to subscribe to, (only filter what is coming in)

Maybe we could have a method 

public IObservable<HassEvent> GetEventObservable(EventType eventType)

It returns an observable and when you subscribe to that IObservable it starts listening to HA with the specified filter

Reply:
No the oposite. I do not want to publish things on the channel unless there is someone listening to it or it will be a memory error / blocking when full. I try to see if I can find another name or solution for it.